src/icp_canister_backend/tests/rewards_tests.rs

```diff
@@ -159,4 +199,649 @@ fn test_reward_distribution_middle_and_final() {
+    let user_a = Principal::from_text("rdmx6-jaaaa-aaaaa-aaadq-cai").unwrap();
+    let user_b = Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap();
+    let malicious_user = Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap();
+
+    let deposit_amount = Nat::from(100_000_000u64); // 1 BTC (assuming BTC @ $100k)
+    let reward_amount = Nat::from(100_000_000u64); // 1 BTC reward pool
+
+    let stakable_episode = get_stakable_episode(&pic, canister_id, 7);
+
+    create_deposit(
+        &pic,
+        canister_id,
+        ledger_id,
+        user_a,
+        deposit_amount.clone(),
+        stakable_episode,
+    );
+    create_deposit(
+        &pic,
+        canister_id,
+        ledger_id,
+        user_b,
+        deposit_amount.clone(),
+        stakable_episode,
+    );
+
+    reward_pool(&pic, canister_id, ledger_id, user_a, reward_amount.clone())
+        .expect("Reward pool should succeed");
+
+    let reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    let last_reward_episode = (reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION;
+    let exact_reward_duration = (last_reward_episode + 1) * EPISODE_DURATION - reward_time;
+    advance_time(&pic, exact_reward_duration);
+
+    let malicious_withdrawal = pic.update_call(
+        canister_id,
+        user_a,
+        "withdraw_rewards",
+        encode_args((vec![1u64],)).unwrap(),
+    );
+
+    let malicious_result =
+        malicious_withdrawal.expect("Malicious withdrawal call should not fail at protocol level");
+    let result: Result<Nat, icp_canister_backend::PoolError> =
+        decode_one(&malicious_result).unwrap();
+    let is_not_owner_error = matches!(result, Err(icp_canister_backend::PoolError::NotOwner));
+    assert!(
+        is_not_owner_error,
+        "Expected NotOwner error, got: {:?}",
+        result
+    );
+
+    let malicious_withdrawal_2 = pic.update_call(
+        canister_id,
+        malicious_user,
+        "withdraw_rewards",
+        encode_args((vec![0u64],)).unwrap(),
+    );
+
+    let malicious_result_2 = malicious_withdrawal_2
+        .expect("Malicious withdrawal call should not fail at protocol level");
+    let result_2: Result<Nat, icp_canister_backend::PoolError> =
+        decode_one(&malicious_result_2).unwrap();
+    let is_not_owner_error_2 = matches!(result_2, Err(icp_canister_backend::PoolError::NotOwner));
+    assert!(
+        is_not_owner_error_2,
+        "Expected NotOwner error for malicious user, got: {:?}",
+        result_2
+    );
+
+    let empty_withdrawal = pic
+        .update_call(
+            canister_id,
+            user_b,
+            "withdraw_rewards",
+            encode_args((Vec::<u64>::new(),)).unwrap(),
+        )
+        .expect("Empty withdrawal call should not fail at protocol level");
+
+    let result: Result<Nat, icp_canister_backend::PoolError> =
+        decode_one(&empty_withdrawal).unwrap();
+    let is_insufficient_balance = matches!(
```

Comment:
What is ??
Why did you randomly decided to use bool variable for this?
Keep things similar to existing code.